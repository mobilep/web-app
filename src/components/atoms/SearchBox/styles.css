.SearchBox {
	display: flex;
	flex-wrap: nowrap;
	border: solid 1px var(--grey4);
	border-radius: 4px;
	background-color: var(--white);
	height: 34px;
	margin-bottom: 15px;
}

.SearchBox svg {
	width: 34px;
	height: 34px;
	color: var(--grey3);
}

.SearchBox-inputBox {
	flex: 0 1 100%;
}

.SearchBox-input {
	font-size: 14px;
	line-height: 34px;
}

.SearchBox-input::-webkit-input-placeholder {
	color: var(--almostBlack);
}

.SearchBox-inputBox .SearchBox-input {
	padding: 0 10px 0 4px;
	background: transparent;
	border: none;
	height: 34px;
	color: var(--almostBlack);
}

.SearchBox-category {
	flex: 1 0 auto;
	padding-left: 5px;
	font-size: 14px;
	line-height: 34px;
	color: var(--grey3);
}

@media (max-width: 768px) {
	.SearchBox {
		border: solid 1px var(--grey2);
		height: 50px;
		align-items: center;
		margin-bottom: 0;
	}

	.SearchBox-input {
		font-size: var(--mobileInputSize);
	}

	.SearchBox-input::placeholder {
		font-size: var(--mobileInputSize);
	}
}
