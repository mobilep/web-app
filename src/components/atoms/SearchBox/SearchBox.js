import React, { Component } from 'react';
import classNames from 'classnames';
import PropTypes from 'prop-types';
import { Subject } from 'rxjs';
import { debounceTime, takeWhile } from 'rxjs/operators';

import { SearchIcon } from '../../../icons';
import { TextInput } from '../../molecules';

import './styles.css';

export default class SearchBox extends Component {

	static propTypes = {
		category: PropTypes.string,
		className: PropTypes.string,
		onChange: PropTypes.func,
		placeholder: PropTypes.string,
		view: PropTypes.string,
	};

	constructor (props) {
		super(props);
		this.createRxSubscription();

		this.state = {
			placeholder: props.placeholder,
			value: '',
			inFocus: false,
		};
	}

	createRxSubscription () {
		this.searchBoxChangeStream$ = new Subject();
		this.searchBoxChangeStream$
			.pipe(
				debounceTime(500),
				takeWhile(() => !this.searchBoxChangeStream$.isStopped)
			)
			.subscribe((value) => {
				this.props.onChange(value);
			});
	}

	componentWillUnmount () {
		this.searchBoxChangeStream$.unsubscribe();
	}

	componentDidUpdate (prevProps) {
		if (prevProps.view !== this.props.view) {
			// eslint-disable-next-line react/no-did-update-set-state
			this.setState({ value: '' });
		}
	}

	getClassName = (className) => classNames('SearchBox', className);

	handleChange = ({ target: { value } }) => {
		this.setState({ value });
		this.searchBoxChangeStream$.next(value);
	};

	handleBlur = () => {
		this.setState({ placeholder: this.props.placeholder, inFocus: false });
	};

	handleFocus = () => {
		this.setState({ placeholder: '', inFocus: true });
	};

	isCategory = () => {
		const { inFocus, value } = this.state;
		const { category } = this.props;

		return category && (inFocus || value);
	};

	render () {
		const { category, className, ...rest } = this.props;
		const { placeholder } = this.state;

		return (
			<div className={this.getClassName(className)}>
				<SearchIcon />
				{
					this.isCategory() &&
					<div className='SearchBox-category'>{category}</div>
				}
				<TextInput
					{ ...rest }
					value={this.state.value}
					className='SearchBox-inputBox'
					inputClassName='SearchBox-input'
					type='text'
					placeholder={placeholder}
					onFocus={this.handleFocus}
					onBlur={this.handleBlur}
					onChange={this.handleChange}
				/>
			</div>
		);
	}
}
