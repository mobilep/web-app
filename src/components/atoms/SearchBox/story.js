import React from 'react';
import { text } from '@storybook/addon-knobs/react';
import { storiesOf } from '@storybook/react';

import SearchBox from './SearchBox';

storiesOf('1. Atoms / SearchBox', module)
	.add('SearchBox', () => <SearchBox name={'somename'} category={text('category', 'To:')} />);
