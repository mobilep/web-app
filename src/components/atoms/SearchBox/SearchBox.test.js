import React from 'react';
import { shallow } from 'enzyme';

import SearchBox from './SearchBox';

describe('Atom - SearchBox', () => {

	it('Should render', () => {
		const wrapper = shallow(<SearchBox name='Name' placeholder='Search' />);
		expect(wrapper).toMatchSnapshot();
	});

	it('Should render with category', () => {
		const wrapper = shallow(<SearchBox name='Name' placeholder='Search' category='Category' />);
		expect(wrapper).toMatchSnapshot();
	});

});
