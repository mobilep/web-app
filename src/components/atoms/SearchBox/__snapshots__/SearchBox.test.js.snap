// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Atom - SearchBox Should render 1`] = `
<div
  className="SearchBox"
>
  <SearchIcon />
  <TextInput
    className="SearchBox-inputBox"
    inputClassName="SearchBox-input"
    name="Name"
    onBlur={[Function]}
    onChange={[Function]}
    onFocus={[Function]}
    placeholder="Search"
    type="text"
    value=""
  />
</div>
`;

exports[`Atom - SearchBox Should render with category 1`] = `
<div
  className="SearchBox"
>
  <SearchIcon />
  <TextInput
    className="SearchBox-inputBox"
    inputClassName="SearchBox-input"
    name="Name"
    onBlur={[Function]}
    onChange={[Function]}
    onFocus={[Function]}
    placeholder="Search"
    type="text"
    value=""
  />
</div>
`;
